Update VERIFICATION.md (#2337)
runtimeverification/evm-semantics: VERIFICATION.md

Verification Instructions for KEVM
==================================

> If you're trying to run Solidity programs with symbolic execution, check out our guide for KONTROL at [docs.runtimeverification.com/kontrol].

Reviewer: ```suggestion
If you're trying to analyze Solidity smart contracts with symbolic execution, check out our guide for Kontrol at [docs.runtimeverification.com/kontrol].
```

The first step is kompiling the `.k` file with the below command.

```sh
kevm kompile-spec sum-to-n-spec.k --target haskell --syntax-module VERIFICATION --main-module VERIFICATION --output-definition sum-to-n-spec/haskell
```

In this example, the arguments used are:

  - `--target haskell`: specify which symbolic backend to use for reasoning.
  - `--syntax-module VERIFICATION`: explicitly state the syntax module.
  - `--main-module VERIFICATION`: explicitly state the main module.
  - `--output-definition sum-to-n-spec/haskell`: the path where the kompiled definition is stored.

Next, run the prover with:

```sh
kevm prove-legacy sum-to-n-spec.k --definition sum-to-n-spec/haskell
```

The expected output is `#Top` which represents that all the claims have been proven.

Debugging a proof
-----------------

For `kevm prove-legacy`, you can use the `--debugger` flag to debug a proof. With it, you can use `step`/`stepf` to navigate through the rewrite steps and `konfig` to display the K configuration. You can see all the available commands using `help`.

> For `kevm prove`. you can use `kevm show-kcfg ...` or `kevm view-kcfg ...` to get a visualization.

Reviewer: ```suggestion
For `kevm prove`, you can use `kevm show-kcfg ...` or `kevm view-kcfg ...` to get a visualization.
```

***Note:*** this is not compatible with `kevm prove-legacy`.

`kevm view-kcfg [spec_file] [--save-directory save_directory] [--claim claim_label] ...` command takes the same basic arguments as `kevm prove ...` does, including:

>   - `spec_file` is the file to look in for specifications. This is the same file used for `kevm prove …`.

Reviewer: ```suggestion
  - `spec_file` is the file to look in for specifications. This is the same file that is used for `kevm prove …`.
```
